apps/drec-ui/src/apps/device-group/logic/smartMeter/chart/useGenerateChartDataset.ts

```diff
@@ -1,9 +1,27 @@
 import { AggregatedReadDTO } from '@energyweb/origin-drec-api-client';
 import { useTheme } from '@mui/material';
+import dayjs from 'dayjs';
+import { DayJsTimeMultiplier } from '../../../../../utils';
 
-export const useGenerateChartDataset = (reads: AggregatedReadDTO[]) => {
+export const useGenerateChartDataset = (
+    reads: AggregatedReadDTO[],
+    start: Date,
+    end: Date,
+    multiplier: DayJsTimeMultiplier,
+    format: string
+) => {
     const theme = useTheme();
-    const currentData = reads.map((read) => read.value);
+    const currentData: number[] = [];
+
+    for (let current = start; current < end; current = dayjs(current).add(1, multiplier).toDate()) {
+        const currentRead = reads.find((read) => {
+            return (
+                dayjs(current).format(format) >= dayjs.utc(read.start).format(format) &&
+                dayjs(current).format(format) < dayjs.utc(read.stop).format(format)
```

Comment:
`dayjs` has built-in utilities for this:

`dayjs('2010-10-20').isSameOrAfter('2010-10-19', 'year')`
`dayjs().isBefore('2011-01-01', 'year')`

Reply:
Yes, I tried with isBetween as well but had some issues with the hourly reads. This way I control it better.